### DISCUSSION TITLE
sh: tailwindcss: command not found

### DISCUSSION THREAD

**Discussion by MrDavidson23** [tailwindlabs/tailwindcss#4953] [Help]
I'm trying to use TailwindCSS, but I can't

Everytime I put npx tailwindcss -o tailwind.css, the bash gives me a:  sh: tailwindcss: command not found

The steps I follow are:

1. Init nodejs: npm init
2. Install tailwindcss:
I have had use npm install -D tailwindcss@latest postcss@latest autoprefixer@latest
And also npm i -D tailwindcss
3. Build tailwind:
npx tailwindcss -o tailwind.css
It gives me: sh: tailwindcss: command not found

**Comment by BiboyQG** [comment]
Try to delete your node_module and reinstall all the package.

**Comment by applyfly** [comment]
If your computer is MacOS, run brew install tailwindcss

**Reply by aoua11** [reply]
thank you so much  
it worked

**Reply by hwisunggh** [reply]
it worked too

**Reply by thanmaiaadimulam** [reply]
thanks a lot ,it worked

**Reply by jaivincoder** [reply]
Thanks it worked

**Comment by mass10** [comment]
I thought `tailwindcss` was moved into `@tailwindcss/cli`.

https://tailwindcss.com/docs/installation/tailwind-cli

**Reply by rozsazoltan** [reply]
Related: [Problem installing TailwindCSS after "npx tailwindcss init -p" command - Error: tailwindcss not recognized](https://stackoverflow.com/a/79380522/15167500)

**Comment by royalcreator1** [comment]
npm install -D tailwindcss@3 postcss autoprefixer
npx tailwindcss init -p

It worked after running this command and add
@tailwind base;
@tailwind components;
@tailwind utilities;

in index.css

**Reply by franklinm54** [reply]
Thanks, it worked

**Reply by apolocine** [reply]
merci beaucoup

**Reply by NewIncome** [reply]
I also did it and it worked!!!
Does anybody know if there is any issue with the newer version??
I had  "tailwindcss": "^4.1.0" , and it wasn't working
Now with the above mentioned command now i have  "tailwindcss": "^3.4.17" , and it works like a charm

By the way, thanks @royalcreator1 !!

**Reply by aarzootrehan** [reply]
Yes, this seems to be an issue with the latest tailwindcss. I had the same issue while using tailwindcss": "^4.1.0" , and it wasn't working.
When I changed it to "^3.4.17", it started to work.

**Reply by thisinnocence** [reply]
> Yes, this seems to be an issue with the latest tailwindcss. I had the same issue while using tailwindcss": "^4.1.0" , and it wasn't working. When I changed it to "^3.4.17", it started to work.

YES!!!

**Reply by Brianoyaro** [reply]
It works like a charm after spending countless days debugging and going through the documentations. Thank you very much.

**Reply by diasgsputra** [reply]
thanks bro

**Reply by rozsazoltan** [reply]
Related: [Problem installing TailwindCSS after "npx tailwindcss init -p" command - Error: tailwindcss not recognized](https://stackoverflow.com/a/79380522/15167500)

**Reply by abuhuraira77788** [reply]
OMG!!! 
i swear after spending in hours debugging, i feel like you're an angel.

**Reply by carlosgarita** [reply]
Two LLMs couldn't give me this solution. And they tried everything. Faith in classic human interaction and classic internet restored. Thank you.

**Reply by fetahokey** [reply]
thanks 👍

**Comment by gocs** [comment]
# Perhaps **VERSION 3** is what you need

my initial setup was just React 19 in Webpack, and i'm not going to install tailwind globally

1. install Tailwind CSS

```sh
npm install -D tailwindcss@3 postcss autoprefixer
npx tailwindcss init -p
```

3. Configure your template paths

make sure your content lists where your template files are. for me it is on src and it should detect the `.js`, or `.jsx` files

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: ["./src/**/*.{js,jsx}"],
  theme: {
    extend: {},
  },
  plugins: [],
};

```

4. Add the Tailwind directives to your CSS

add these lines at the top of your `index.css` file

```css
@tailwind base;
@tailwind components;
@tailwind utilities;
```

6. Import the css file

my `index.css` was already imported within `index.js`

4. start your build process

```sh
npm start
```

7. Start using Tailwind in your project

```jsx
export default function App() {
  return (
    <h1 className="text-3xl font-bold underline">
      Hello world!
    </h1>
  )
}
```
